client/gutenberg/extensions/giphy/edit.js

```diff
@@ -0,0 +1,185 @@
+		const split = array.split( delimiter );
+		return split[ split.length - 1 ];
+	};
+
+	fetch = url => {
+		const { setAttributes } = this.props;
+		const xhr = new XMLHttpRequest();
+		xhr.open( 'GET', url );
+		xhr.onload = () => {
+			if ( xhr.status === 200 ) {
+				const res = JSON.parse( xhr.responseText );
+				const giphyData = res.data.length > 0 ? res.data[ 0 ] : res.data;
+				// No results
+				if ( ! giphyData.images ) {
+					return;
+				}
+				const topPadding = Math.floor(
+					( giphyData.images.original.height / giphyData.images.original.width ) * 100
+				);
+				const giphyUrl = giphyData.embed_url;
+				setAttributes( { giphyUrl, topPadding } );
+				this.maintainFocus( 500 );
+			} else {
+				// Error handling TK
+			}
+		};
+		xhr.send();
+	};
+
+	setFocus = () => {
+		this.maintainFocus();
+		this.textControlRef.current.querySelector( 'input' ).focus();
+	};
+
+	maintainFocus = ( timeoutDuration = 3500 ) => {
+		this.setState( { focus: true }, () => {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+			if ( this.hasSearchText() ) {
+				this.timer = setTimeout( () => {
+					this.setState( { focus: false } );
+				}, timeoutDuration );
+			}
+		} );
+	};
+
+	clearFocus = () => {
+		this.setState( { focus: false }, () => {
+			if ( this.timer ) {
+				clearTimeout( this.timer );
+			}
+		} );
+	};
+
+	hasSearchText = () => {
+		const { attributes } = this.props;
+		const { searchText } = attributes;
+		return searchText && searchText.length > 0;
+	};
+
+	render() {
+		const { attributes, className, isSelected, setAttributes } = this.props;
+		const { align, caption, giphyUrl, searchText, topPadding } = attributes;
+		const { focus } = this.state;
+		const style = {
+			paddingTop: `${ topPadding }%`,
+		};
+		const classes = classNames( className, `align${ align }` );
+		const textControlClasses = classNames(
+			'wp-block-jetpack-giphy_text-input-field',
+			focus || ! this.hasSearchText() ? 'has-focus' : 'no-focus'
+		);
+		return (
+			<div className={ classes }>
+				<figure style={ style }>
+					<div
+						className="wp-block-jetpack-giphy_cover"
+						onClick={ this.setFocus }
+						onKeyDown={ this.setFocus }
+						ref={ this.textControlRef }
+						role="button"
+						tabIndex="0"
+					>
+						{ ( ! searchText || isSelected ) && (
+							<TextControl
+								className={ textControlClasses }
+								label={ __( 'Search or paste a Giphy URL' ) }
+								placeholder={ __( 'Search or paste a Giphy URL' ) }
+								onChange={ this.onSearchTextChange }
+								onClick={ this.maintainFocus }
+								value={ searchText }
+							/>
+						) }
+					</div>
+					<iframe src={ giphyUrl } title={ searchText } />
+				</figure>
+				{ ( ! RichText.isEmpty( caption ) || isSelected ) && (
+					<RichText
+						className="caption"
+						inlineToolbar
+						onChange={ value => setAttributes( { caption: value } ) }
+						placeholder={ __( 'Write caption…' ) }
+						tagName="figcaption"
```

Comment:
`figcaption` is only permitted as the first or last direct descendent of `figure`:

> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption

We may need to restructure this a bit.

Reply:
Before addressing this, perhaps we should consider removing the caption as @aldavigdis asks here: https://github.com/Automattic/wp-calypso/pull/29638#issuecomment-454818106. @scruffian what do you think? Caption or no caption? The Atavist block does have a caption, although I'll be able to work around this in migrations by converting an Atavist Giphy block to Gutenberg Giphy + Gutenberg Paragraph (or Heading).